### PR TITLE
Nsng 567 fe finetuning

### PR DESCRIPTION
**Pull request by igor-kamil**

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -37,5 +37,7 @@ export default {
       },
     },
   },
-  plugins: [],
+  plugins: [
+    require('@tailwindcss/typography'),
```
Comment (FrantisekMichalSebestyen) [inline tailwind.config.js:41]: Could you explain what is the use case here?

File: pages/index.vue [hunk 2]
```diff
@@ -333,46 +367,50 @@ const closePopover = () => {
...
+
+watchEffect(() => {
+  if (!openedPopover.value || !sectionsData.value) return;
+  openedPopover.value = sectionsData.value.find(
+    (section) => openedPopover.value.id === section.id,
+  );
+});
+
+const calculateSquareDimension = () => {
+  const viewportHeight = window.innerHeight;
```
Comment (FrantisekMichalSebestyen) [inline pages/index.vue:404]: We need to rework this, so we dont use window -> so we can render this on server
+ pxValue is quite ambivalent name

**Review by FrantisekMichalSebestyen — CHANGES_REQUESTED** [review]
Just marked few things:
- we should split sections from items on api
- display items based on the span-x and span-y order 
- also changes to editor are desirable

File: tailwind.config.js (hunk 1 above)
Comment (igor-kamil) [reply tailwind.config.js:41]: > The official Tailwind CSS Typography plugin provides a set of prose classes you can use to add beautiful typographic defaults to any vanilla HTML you don’t control, like HTML rendered from Markdown

we use it also elsewhere to set typographic behaviour on data parsed from markdown.
the default `prose` class already included proposed sizing, line-height and spacing.
I would prefer to use this package as it sets good standard for text in paragraphs. but the truth is that if necessary, the same can be achieved without the plugin. if you find this such a case @FrantisekMichalSebestyen , feel free to implement it in standalone task.

**Comment by igor-kamil** [comment]
> Just marked few things:
> 
> * we should split sections from items on api

if you talk about avoiding the delay after switch language - I think turning on cache or SSR is better solution 

> * display items based on the span-x and span-y order

makes sense. but fixing this in admin takes just few minutes. and saves our time 😁 

> * also changes to editor are desirable

I agree 😉 

suggestions makes sense. but imho they are outside of the scope of this task and I suggest creating new tasks for them

File: assets/img/info.svg [hunk 3]
```diff
@@ -1,4 +1,7 @@
-<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256">
-    <path d="M128,24A104,104,0,1,0,232,128,104.11,104.11,0,0,0,128,24Zm0,192a88,88,0,1,1,88-88A88.1,88.1,0,0,1,128,216Zm16-40a8,8,0,0,1-8,8,16,16,0,0,1-16-16V128a8,8,0,0,1,0-16,16,16,0,0,1,16,16v40A8,8,0,0,1,144,176ZM112,84a12,12,0,1,1,12,12A12,12,0,0,1,112,84Z">
-    </path>
-</svg>
\ No newline at end of file
+<svg viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (FrantisekMichalSebestyen) [inline assets/img/info.svg:1]: @igor-kamil this should come from https://phosphoricons.com/ + fill and stroke should be set from outside

File: assets/img/logo.svg [hunk 4]
```diff
@@ -1,3 +1,3 @@
-<svg viewBox="0 0 42 42" fill="none" xmlns="http://www.w3.org/2000/svg">
-<path d="M25.103 31.7737H38.1748V38.1445H22.4777V22.4777H38.1748V28.8485H25.103H25.003V28.9485V31.6737V31.7737H25.103ZM1 0.9H0.9V1V22.3777V22.4777H1H19.5223V38.1748H3.82521V25.103V25.003H3.72521H1H0.9V25.103V41V41.1H1H41H41.1V41V1V0.9H41H1ZM22.4777 19.5223V5.90456L36.0954 19.5223H22.4777ZM3.82521 3.82521H19.5223V19.5223H3.82521V3.82521ZM38.1748 17.4429L24.5571 3.82521H38.1748V17.4429Z" fill="white" stroke="white" stroke-width="0.2"/>
+<svg viewBox="0 0 42 42" xmlns="http://www.w3.org/2000/svg">
+<path d="M25.103 31.7737H38.1748V38.1445H22.4777V22.4777H38.1748V28.8485H25.103H25.003V28.9485V31.6737V31.7737H25.103ZM1 0.9H0.9V1V22.3777V22.4777H1H19.5223V38.1748H3.82521V25.103V25.003H3.72521H1H0.9V25.103V41V41.1H1H41H41.1V41V1V0.9H41H1ZM22.4777 19.5223V5.90456L36.0954 19.5223H22.4777ZM3.82521 3.82521H19.5223V19.5223H3.82521V3.82521ZM38.1748 17.4429L24.5571 3.82521H38.1748V17.4429Z" fill="currentColor" stroke="currentColor" stroke-width="0.2"/>
```
Comment (FrantisekMichalSebestyen) [inline assets/img/logo.svg:2]: 👍

File: pages/index.vue [hunk 5]
```diff
@@ -156,91 +217,119 @@
   >
     <div
       v-if="openedZoomId"
-      class="absolute bottom-0 left-0 z-20 h-[95%] w-2/3 border-t-2 border-t-black bg-white"
+      class="absolute bottom-0 left-0 top-[80px] z-20 w-[calc(100vw-528px)] border-t-2 border-t-black bg-white"
```
Comment (FrantisekMichalSebestyen) [inline pages/index.vue:220]:
```suggestion
      class="absolute bottom-0 left-0 top-20 z-20 w-[calc(100vw-528px)] border-t-2 border-t-black bg-white"
```

File: pages/index.vue [hunk 6]
```diff
@@ -156,91 +217,119 @@
...
+          class="prose-xl leading-8"
+          v-html="openedPopover.text"
+        ></article>
       </div>
     </template>
   </Popover>
 </template>
 <script setup>
-import { watchEffect } from "vue";
+import { watch } from "vue";
```
Comment (FrantisekMichalSebestyen) [inline pages/index.vue:297]:
```suggestion
```

File: pages/index.vue [hunk 7]
```diff
@@ -156,91 +217,119 @@
...
 import Logo from "~/assets/img/logo.svg?component";
 import Info from "~/assets/img/info.svg?component";
 import { NUM_OF_COLUMNS, NUM_OF_ROWS, SQUARE_DIMENSION } from "../consts";
 
 const openedPopover = ref(null);
 const openedZoomId = ref(null);
+const maxGridScrollValue = 1000;
```
Comment (FrantisekMichalSebestyen) [inline pages/index.vue:304]: move to consts?

File: pages/index.vue (hunk 2 above)
Comment (FrantisekMichalSebestyen) [reply pages/index.vue:410]: did you create a task for this?

File: pages/index.vue [hunk 8]
```diff
@@ -290,41 +372,51 @@ const closePopover = () => {
...
+  );
+});
+
+const calculateSquareDimension = () => {
+  const viewportHeight = window.innerHeight;
+  const pxValue = 110;
+  return (viewportHeight - pxValue) / NUM_OF_ROWS;
+};
+
+const translateShadowStyle = computed(() => {
```
Comment (FrantisekMichalSebestyen) [inline pages/index.vue:415]: this is weird as welll. Please create a task/fix now

File: pages/index.vue [hunk 9]
```diff
@@ -290,41 +372,51 @@ const closePopover = () => {
...
+} = useFetch(`${apiUrl}/sections`, {
+  headers: {
+    "Accept-Language": locale,
   },
-];
+  watch: [locale],
+});
+
+// increment all x/y positions +1 to fix issues with grid positioning
+const sectionsData = computed(() => {
```
Comment (FrantisekMichalSebestyen) [inline pages/index.vue:390]: task/fix

**Review by FrantisekMichalSebestyen — CHANGES_REQUESTED** [review]

File: pages/index.vue (hunk 2 above)
Comment (igor-kamil) [reply pages/index.vue:408]: https://jira.sng.sk/browse/NSNG-577

File: pages/index.vue (hunk 9 above)
Comment (igor-kamil) [reply pages/index.vue:388]: https://jira.sng.sk/browse/NSNG-579

**Pull request merged**